From ec9cd39d1bd58ea11e9442d4d6149f108b7ecc99 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 15 Oct 2024 19:51:47 +0200
Subject: [PATCH 01/14] libslic3r: Add missing include for tbb::spin_mutex
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/libslic3r/Brim.cpp: In function ‘Slic3r::ExPolygons Slic3r::outer_inner_brim_area(const Print&, float, std::map<ObjectID, std::vector<ExPolygon> >&, std::map<ObjectID, std::vector<ExPolygon> >&, std::vector<std::pair<ObjectID, unsigned int> >&, std::vector<unsigned int>&)’:
src/libslic3r/Brim.cpp:1084:18: error: ‘spin_mutex’ is not a member of ‘tbb’
 1084 |             tbb::spin_mutex brimMutex;
      |                  ^~~~~~~~~~
---
 src/libslic3r/Brim.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libslic3r/Brim.cpp b/src/libslic3r/Brim.cpp
index 57683a74cc8a..f9c0411b23a3 100644
--- a/src/libslic3r/Brim.cpp
+++ b/src/libslic3r/Brim.cpp
@@ -12,6 +12,7 @@
 #include <numeric>
 #include <unordered_set>
 #include <tbb/parallel_for.h>
+#include <tbb/spin_mutex.h>
 
 #include <boost/log/trivial.hpp>
 
-- 
2.47.0


From 86a6e10e341da8f7184fa61916c9e80b843fa495 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 15 Oct 2024 21:34:19 +0200
Subject: [PATCH 02/14] libslic3r: Fix declaration in FlushVolPredictor.hpp
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In file included from src/libslic3r/FlushVolPredictor.cpp:1:
src/libslic3r/FlushVolPredictor.hpp:44:34: error: ‘string’ in namespace ‘std’ does not name a type
   44 |     FlushVolPredictor(const std::string& data_file);
      |                                  ^~~~~~

src/libslic3r/FlushVolPredictor.hpp:49:5: error: ‘uint64_t’ does not name a type
   49 |     uint64_t generate_hash_key(const RGB& from, const RGB& to);
      |     ^~~~~~~~

In file included from src/libslic3r/FlushVolPredictor.cpp:1:
src/libslic3r/FlushVolPredictor.hpp:44:34: error: ‘string’ in namespace ‘std’ does not name a type
   44 |     FlushVolPredictor(const std::string& data_file);
      |                                  ^~~~~~
---
 src/libslic3r/FlushVolPredictor.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libslic3r/FlushVolPredictor.hpp b/src/libslic3r/FlushVolPredictor.hpp
index 0054c70dd8a6..378a608fedb2 100644
--- a/src/libslic3r/FlushVolPredictor.hpp
+++ b/src/libslic3r/FlushVolPredictor.hpp
@@ -1,7 +1,10 @@
 #ifndef FLUSH_VOL_PREDICTOR_HPP
 #define FLUSH_VOL_PREDICTOR_HPP
 
+#include<cstdint>
 #include<unordered_map>
+#include<string>
+#include<vector>
 
 namespace FlushPredict
 {
-- 
2.47.0


From 3ce3ea83faeb9752924b92aca4d5497e7b21d803 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 15 Oct 2024 21:39:13 +0200
Subject: [PATCH 03/14] libslic3r: Fix format calls
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/libslic3r/Support/TreeSupport.cpp:3204:41: error: ‘format’ was not declared in this scope; did you mean ‘boost::format’?
 3204 |             BOOST_LOG_TRIVIAL(trace) << format("plan_layer_heights0 (%.2f,%.2f)->(%.2f,%.2f): ", z1, h1, z2, h2);
      |                                         ^~~~~~
      |                                         boost::format
---
 src/libslic3r/Support/TreeSupport.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libslic3r/Support/TreeSupport.cpp b/src/libslic3r/Support/TreeSupport.cpp
index a0c4ab81905b..71b734a4425c 100644
--- a/src/libslic3r/Support/TreeSupport.cpp
+++ b/src/libslic3r/Support/TreeSupport.cpp
@@ -2,6 +2,7 @@
 #include <chrono>
 #include <math.h>
 
+#include "format.hpp"
 #include "ClipperUtils.hpp"
 #include "Fill/FillBase.hpp"
 #include "I18N.hpp"
-- 
2.47.0


From efff0d40c8507532fee999b68a1b5f8ec8717205 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 15 Oct 2024 21:40:59 +0200
Subject: [PATCH 04/14] slic3r: Fix missing declarations in wxMediaCtrl3.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/wxMediaCtrl3.h:80:10: error: ‘condition_variable’ in namespace ‘std’ does not name a type
   80 |     std::condition_variable m_cond;
      |          ^~~~~~~~~~~~~~~~~~
src/slic3r/GUI/wxMediaCtrl3.h:27:1: note: ‘std::condition_variable’ is defined in header ‘<condition_variable>’; did you forget to ‘#include <condition_variable>’?
   26 | #include "Printer/BambuTunnel.h"
  +++ |+#include <condition_variable>
   27 |
src/slic3r/GUI/wxMediaCtrl3.h:81:10: error: ‘thread’ in namespace ‘std’ does not name a type
   81 |     std::thread m_thread;
      |          ^~~~~~
src/slic3r/GUI/wxMediaCtrl3.h:27:1: note: ‘std::thread’ is defined in header ‘<thread>’; did you forget to ‘#include <thread>’?
   26 | #include "Printer/BambuTunnel.h"
  +++ |+#include <thread>
   27 |

In file included from src/slic3r/GUI/MediaPlayCtrl.h:17,
                 from src/slic3r/GUI/MediaPlayCtrl.cpp:1:
src/slic3r/GUI/wxMediaCtrl3.h:77:13: error: field ‘m_frame’ has incomplete type ‘wxImage’
   77 |     wxImage m_frame;
      |             ^~~~~~~
---
 src/slic3r/GUI/wxMediaCtrl3.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/slic3r/GUI/wxMediaCtrl3.h b/src/slic3r/GUI/wxMediaCtrl3.h
index d75dff9b5c37..86bc7755ea86 100644
--- a/src/slic3r/GUI/wxMediaCtrl3.h
+++ b/src/slic3r/GUI/wxMediaCtrl3.h
@@ -23,6 +23,11 @@ void wxMediaCtrl_OnSize(wxWindow * ctrl, wxSize const & videoSize, int width, in
 #else
 
 #define BAMBU_DYNAMIC
+#include <condition_variable>
+#include <thread>
+#ifndef _WIN32
+#include <wx/image.h>
+#endif
 #include "Printer/BambuTunnel.h"
 
 class AVVideoDecoder;
-- 
2.47.0


From 6144d993a468916e237e91ef7903f7a5dde46361 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 15 Oct 2024 22:42:39 +0200
Subject: [PATCH 05/14] slic3r: Fix wxDisplay declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp: In member function ‘virtual void Slic3r::GUI::GLGizmoFlatten::on_render_input_window(float, float, float)’:
src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp:45:27: error: ‘wxDisplay’ was not declared in this scope; did you mean ‘wxGetDisplay’?
   45 |     double screen_scale = wxDisplay(wxGetApp().plater()).GetScaleFactor();
      |                           ^~~~~~~~~
      |                           wxGetDisplay
---
 src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp b/src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp
index ab174c39a17d..c02b64df6902 100644
--- a/src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp
+++ b/src/slic3r/GUI/Gizmos/GLGizmoFlatten.cpp
@@ -7,6 +7,8 @@
 #include "libslic3r/Geometry/ConvexHull.hpp"
 #include "libslic3r/Model.hpp"
 
+#include <wx/display.h>
+
 #include <numeric>
 #include <imgui/imgui_internal.h>
 #include <GL/glew.h>
-- 
2.47.0


From ca86e2ac87edcc68ff3413a8677d3844171f30e1 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 15 Oct 2024 22:44:39 +0200
Subject: [PATCH 06/14] slic3r: Fix boost::nowide declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/BitmapCache.cpp: In static member function ‘static NSVGimage* Slic3r::GUI::BitmapCache::nsvgParseFromFileWithReplace(const char*, const char*, float, const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >&)’:
src/slic3r/GUI/BitmapCache.cpp:279:17: error: ‘boost::nowide’ has not been declared
  279 |     fp = boost::nowide::fopen(filename, "rb");
      |                 ^~~~~~
---
 src/slic3r/GUI/BitmapCache.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/slic3r/GUI/BitmapCache.cpp b/src/slic3r/GUI/BitmapCache.cpp
index 788681a3a715..8e5990c892a9 100644
--- a/src/slic3r/GUI/BitmapCache.cpp
+++ b/src/slic3r/GUI/BitmapCache.cpp
@@ -6,6 +6,7 @@
 #include "GUI_Utils.hpp"
 
 #include <boost/filesystem.hpp>
+#include <boost/nowide/cstdio.hpp>
 
 #ifdef __WXGTK2__
     // Broken alpha workaround
-- 
2.47.0


From dcbd217a33e2473895519ffaa78d9f11a9cf6e5d Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Oct 2024 13:16:30 +0200
Subject: [PATCH 07/14] slic3r: Fix missing boost::lexical_cast declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/MediaFilePanel.cpp: In lambda function:
src/slic3r/GUI/MediaFilePanel.cpp:508:49: error: ‘lexical_cast’ is not a member of ‘boost’
  508 |                 url += "&refresh_url=" + boost::lexical_cast<std::string>(&refresh_agora_url);
      |                                                 ^~~~~~~~~~~~
---
 src/slic3r/GUI/MediaFilePanel.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/slic3r/GUI/MediaFilePanel.cpp b/src/slic3r/GUI/MediaFilePanel.cpp
index c8256ae8c421..aa6a79c1bd63 100644
--- a/src/slic3r/GUI/MediaFilePanel.cpp
+++ b/src/slic3r/GUI/MediaFilePanel.cpp
@@ -9,6 +9,7 @@
 #include "Printer/PrinterFileSystem.h"
 #include "MsgDialog.hpp"
 #include "Widgets/ProgressDialog.hpp"
+#include <boost/lexical_cast.hpp>
 #include <libslic3r/Model.hpp>
 #include <libslic3r/Format/bbs_3mf.hpp>
 
-- 
2.47.0


From 4ea2b87b90e455377ec0ff16f88cb23ea24eebb2 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Oct 2024 13:18:18 +0200
Subject: [PATCH 08/14] slic3r: Fix missing union_ex/diff_ex declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp: In member function ‘void Slic3r::GUI::GLGizmoBrimEars::first_layer_slicer()’:
src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp:870:27: error: ‘union_ex’ was not declared in this scope; did you mean ‘union’?
  870 |                 part_ex = union_ex(part_ex, sliced_layer);
      |                           ^~~~~~~~
      |                           union
src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp:876:21: error: ‘diff_ex’ was not declared in this scope
  876 |     m_first_layer = diff_ex(part_ex, negative_ex);
      |                     ^~~~~~~
---
 src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp b/src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp
index dac6a4ec7c3b..51feecff22dc 100644
--- a/src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp
+++ b/src/slic3r/GUI/Gizmos/GLGizmoBrimEars.cpp
@@ -5,6 +5,7 @@
 #include "slic3r/GUI/Gizmos/GLGizmosCommon.hpp"
 #include "slic3r/GUI/GUI_App.hpp"
 #include "slic3r/GUI/Plater.hpp"
+#include "libslic3r/ClipperUtils.hpp"
 #include "libslic3r/ExPolygon.hpp"
 
 namespace Slic3r { namespace GUI {
-- 
2.47.0


From 0a98f8674782cb1ac2383fe375d3454a3fb07629 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Oct 2024 17:24:19 +0200
Subject: [PATCH 09/14] libslic3r: Fix Model declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/libslic3r/Format/STEP.hpp:39:41: error: ‘Model’ has not been declared
   39 | extern bool load_step(const char *path, Model *model,
      |                                         ^~~~~
---
 src/libslic3r/Format/STEP.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libslic3r/Format/STEP.hpp b/src/libslic3r/Format/STEP.hpp
index 397ee2c1bba2..71453b3fc9dd 100644
--- a/src/libslic3r/Format/STEP.hpp
+++ b/src/libslic3r/Format/STEP.hpp
@@ -12,6 +12,7 @@ namespace fs = boost::filesystem;
 
 namespace Slic3r {
 
+class Model;
 class TriangleMesh;
 class ModelObject;
 
-- 
2.47.0


From 05f422c3e09a7a6def4a26a7c2d7c371fd279f53 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Oct 2024 17:25:09 +0200
Subject: [PATCH 10/14] slic3r: Fix std::future declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/StepMeshDialog.hpp:40:10: error: ‘future’ in namespace ‘std’ does not name a template type
   40 |     std::future<unsigned int> task;
      |          ^~~~~~
src/slic3r/GUI/StepMeshDialog.hpp:8:1: note: ‘std::future’ is defined in header ‘<future>’; did you forget to ‘#include <future>’?
    7 | #include "Widgets/Button.hpp"
  +++ |+#include <future>
    8 | class Button;
---
 src/slic3r/GUI/StepMeshDialog.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/slic3r/GUI/StepMeshDialog.hpp b/src/slic3r/GUI/StepMeshDialog.hpp
index 48bff2adc35f..df2175d36804 100644
--- a/src/slic3r/GUI/StepMeshDialog.hpp
+++ b/src/slic3r/GUI/StepMeshDialog.hpp
@@ -1,6 +1,7 @@
 #ifndef _STEP_MESH_DIALOG_H_
 #define _STEP_MESH_DIALOG_H_
 
+#include <future>
 #include <thread>
 #include "GUI_Utils.hpp"
 #include "libslic3r/Format/STEP.hpp"
@@ -44,4 +45,4 @@ private:
     void stop_task();
 };
 
-#endif  // _STEP_MESH_DIALOG_H_
\ No newline at end of file
+#endif  // _STEP_MESH_DIALOG_H_
-- 
2.47.0


From dac14d026cfe76c1fcb2dbde72a0a2abc1783fe1 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Oct 2024 22:59:30 +0200
Subject: [PATCH 11/14] slic3r: Fix missing includes in AVVideoDecoder
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In file included from src/slic3r/GUI/AVVideoDecoder.cpp:1:
src/slic3r/GUI/AVVideoDecoder.hpp:28:20: error: ‘wxImage’ has not been declared
   28 |     bool toWxImage(wxImage &image, wxSize const &size);
      |                    ^~~~~~~
src/slic3r/GUI/AVVideoDecoder.hpp:28:36: error: ‘wxSize’ has not been declared
   28 |     bool toWxImage(wxImage &image, wxSize const &size);
      |                                    ^~~~~~
src/slic3r/GUI/AVVideoDecoder.hpp:38:10: error: ‘vector’ in namespace ‘std’ does not name a template type
   38 |     std::vector<uint8_t> bits_;
      |          ^~~~~~
src/slic3r/GUI/AVVideoDecoder.hpp:9:1: note: ‘std::vector’ is defined in header ‘<vector>’; did you forget to ‘#include <vector>’?
    8 |     #include <libswscale/swscale.h>
  +++ |+#include <vector>
    9 | }

src/slic3r/GUI/AVVideoDecoder.cpp:145:89: error: invalid use of incomplete type ‘class wxBitmap’
  145 |     bitmap = wxBitmap((char const *) bits_.data(), size.GetWidth(), size.GetHeight(), 32);
      |                                                                                         ^
---
 src/slic3r/GUI/AVVideoDecoder.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/slic3r/GUI/AVVideoDecoder.hpp b/src/slic3r/GUI/AVVideoDecoder.hpp
index 239352aeaa74..4111e860a210 100644
--- a/src/slic3r/GUI/AVVideoDecoder.hpp
+++ b/src/slic3r/GUI/AVVideoDecoder.hpp
@@ -7,6 +7,11 @@ extern "C" {
     #include <libavcodec/avcodec.h>
     #include <libswscale/swscale.h>
 }
+#include <vector>
+#include <wx/bitmap.h>
+#include <wx/gdicmn.h>
+#include <wx/image.h>
+
 class wxBitmap;
 
 class AVVideoDecoder
-- 
2.47.0


From 600b08d6d88ac7c1d824e580c4e682a82d8e9b71 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Oct 2024 23:04:43 +0200
Subject: [PATCH 12/14] slic3r: Fix missing includes in wxMediaCtrl2
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/wxMediaCtrl2.cpp: In lambda function:
src/slic3r/GUI/wxMediaCtrl2.cpp:170:13: error: ‘wxMessageBox’ was not declared in this scope; did you mean ‘wxInfoMessageBox’?
  170 |             wxMessageBox(_L("Your system is missing H.264 codecs for GStreamer, which are required to play video.  (Try installing the gstreamer1.0-plugins-bad or gstreamer1.0-libav packages, then restart Bambu Studio?)"), _L("Error"), wxOK);
      |             ^~~~~~~~~~~~
      |             wxInfoMessageBox
src/slic3r/GUI/wxMediaCtrl2.cpp: In member function ‘void wxMediaCtrl2::Load(wxURI)’:
src/slic3r/GUI/wxMediaCtrl2.cpp:179:5: error: ‘wxLog’ has not been declared
  179 |     wxLog::EnableLogging(false);
      |     ^~~~~
---
 src/slic3r/GUI/wxMediaCtrl2.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/slic3r/GUI/wxMediaCtrl2.cpp b/src/slic3r/GUI/wxMediaCtrl2.cpp
index 82876cf33fbf..4a9cba4e808b 100644
--- a/src/slic3r/GUI/wxMediaCtrl2.cpp
+++ b/src/slic3r/GUI/wxMediaCtrl2.cpp
@@ -1,6 +1,8 @@
 #include "wxMediaCtrl2.h"
 #include "I18N.hpp"
 #include "libslic3r/Utils.hpp"
+#include <wx/log.h>
+#include <wx/msgdlg.h>
 #ifdef __WIN32__
 #include <versionhelpers.h>
 #include <wx/msw/registry.h>
-- 
2.47.0


From 1bd11798384de90e92ca5871c7a3f82ce406283a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 17 Oct 2024 11:30:14 +0200
Subject: [PATCH 13/14] slic3r: Fix missing wxPaintDC declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/wxMediaCtrl3.cpp: In member function ‘void wxMediaCtrl3::paintEvent(wxPaintEvent&)’:
src/slic3r/GUI/wxMediaCtrl3.cpp:121:5: error: ‘wxPaintDC’ was not declared in this scope; did you mean ‘wxPoint’?
  121 |     wxPaintDC dc(this);
      |     ^~~~~~~~~
      |     wxPoint
---
 src/slic3r/GUI/wxMediaCtrl3.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/slic3r/GUI/wxMediaCtrl3.cpp b/src/slic3r/GUI/wxMediaCtrl3.cpp
index dc91b8525c02..acd04beef78e 100644
--- a/src/slic3r/GUI/wxMediaCtrl3.cpp
+++ b/src/slic3r/GUI/wxMediaCtrl3.cpp
@@ -2,6 +2,7 @@
 #include "AVVideoDecoder.hpp"
 #include "I18N.hpp"
 #include "libslic3r/Utils.hpp"
+#include <wx/dcclient.h>
 #ifdef __WIN32__
 #include <versionhelpers.h>
 #include <wx/msw/registry.h>
-- 
2.47.0


From 22985a57fee15512541ff3d6924983b31eddb8ef Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 17 Oct 2024 11:31:26 +0200
Subject: [PATCH 14/14] slic3r: Fix missing BOOST_LOG_TRIVIAL declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/wxMediaCtrl3.cpp:181:23: error: ‘info’ was not declared in this scope
  181 |     BOOST_LOG_TRIVIAL(info) << msg.ToUTF8().data();
      |                       ^~~~
src/slic3r/GUI/wxMediaCtrl3.cpp:181:5: error: ‘BOOST_LOG_TRIVIAL’ was not declared in this scope
  181 |     BOOST_LOG_TRIVIAL(info) << msg.ToUTF8().data();
      |     ^~~~~~~~~~~~~~~~~
---
 src/slic3r/GUI/wxMediaCtrl3.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/slic3r/GUI/wxMediaCtrl3.cpp b/src/slic3r/GUI/wxMediaCtrl3.cpp
index acd04beef78e..90950b6b101c 100644
--- a/src/slic3r/GUI/wxMediaCtrl3.cpp
+++ b/src/slic3r/GUI/wxMediaCtrl3.cpp
@@ -2,6 +2,7 @@
 #include "AVVideoDecoder.hpp"
 #include "I18N.hpp"
 #include "libslic3r/Utils.hpp"
+#include <boost/log/trivial.hpp>
 #include <wx/dcclient.h>
 #ifdef __WIN32__
 #include <versionhelpers.h>
-- 
2.47.0

